compio-driver/src/iocp/mod.rs

```diff
@@ -456,28 +456,32 @@ impl RawOp {
         self.op.as_ptr()
     }
 
-    pub fn set_completed(&mut self) -> bool {
-        self.completed = true;
+    pub fn set_cancelled(&mut self) -> bool {
+        self.cancelled = true;
+        self.has_result()
+    }
+
+    pub fn set_result(&mut self, res: io::Result<usize>) -> bool {
+        self.result = Some(res);
         self.cancelled
     }
 
-    pub fn set_cancelled(&mut self) -> bool {
-        self.cancelled = true;
-        self.completed
+    pub fn has_result(&self) -> bool {
+        self.result.is_some()
     }
 
     /// # Safety
     /// The caller should ensure the correct type.
-    pub unsafe fn into_inner<T: OpCode>(self) -> T {
-        let this = ManuallyDrop::new(self);
-        let this: Box<Overlapped<T>> = Box::from_raw(this.op.cast().as_ptr());
-        this.op
+    pub unsafe fn into_inner<T: OpCode>(self) -> BufResult<usize, T> {
```

Comment:
Should this be changed to `BufResult<Option<usize>, T>`? If not, the doc should mention the possibility of panicking.

Reply:
Added docs.